apps/price_pusher/src/evm/evm.ts

```diff
@@ -223,42 +250,121 @@ export class EvmPricePusher implements IPricePusher {
 
     this.logger.debug(`Using gas price: ${gasPrice} and nonce: ${txNonce}`);
 
-    this.pythContract.methods
-      .updatePriceFeedsIfNecessary(
-        priceFeedUpdateData,
-        priceIdsWith0x,
-        pubTimesToPush
-      )
-      .send({
-        value: updateFee,
-        gasPrice,
-        nonce: txNonce,
-        gasLimit: this.gasLimit,
-      })
-      .on("transactionHash", (hash: string) => {
-        this.logger.info({ hash }, "Price update successful");
-      })
-      .on("error", (err: Error, receipt?: TransactionReceipt) => {
-        if (err.message.includes("revert")) {
-          // Since we are using custom error structs on solidity the rejection
-          // doesn't return any information why the call has reverted. Assuming that
-          // the update data is valid there is no possible rejection cause other than
-          // the target chain price being already updated.
+    const pubTimesToPushParam = pubTimesToPush.map((pubTime) =>
+      BigInt(pubTime)
+    );
+
+    try {
+      const { request } =
+        await this.pythContract.simulate.updatePriceFeedsIfNecessary(
+          [priceFeedUpdateData, priceIdsWith0x, pubTimesToPushParam],
+          {
+            value: updateFee,
+            gasPrice: BigInt(Math.round(gasPrice)),
+            nonce: txNonce,
+            gas:
+              this.gasLimit !== undefined
+                ? BigInt(Math.round(this.gasLimit))
+                : undefined,
+          }
+        );
+
+      this.logger.debug({ request }, "Simulated request successfully");
+
+      const hash = await this.pythContract.write.updatePriceFeedsIfNecessary(
```

Comment:
In viem you can use the response of a simulation as the argument to `write` i.e.: https://viem.sh/docs/contract/simulateContract.html#pairing-with-writecontract

Reply:
Unfortunately in contrary to the documentation the simulate call result is not compatible with the type required there, the account type being passed is different. @jxom fyi.